File: src/main/java/com/codingforcookies/betterrecords/common/packets/PacketHandler.java
package com.codingforcookies.betterrecords.common.packets;

import com.codingforcookies.betterrecords.api.connection.RecordConnection;
import net.minecraftforge.fml.common.network.FMLEmbeddedChannel;
import net.minecraftforge.fml.common.network.FMLOutboundHandler;
import net.minecraftforge.fml.relauncher.Side;

import java.util.EnumMap;

public class PacketHandler{

    public static EnumMap<Side, FMLEmbeddedChannel> channels;

    public static void sendWireConnectionFromClient(RecordConnection connection){
        channels.get(Side.CLIENT).attr(FMLOutboundHandler.FML_MESSAGETARGET).set(FMLOutboundHandler.OutboundTarget.TOSERVER);
        channels.get(Side.CLIENT).writeOutbound(new PacketWireConnection(connection));
    }

    public static void sendURLWriteFromClient(int x, int y, int z, String name, String url, String localName, int size, int color, String author){
        channels.get(Side.CLIENT).attr(FMLOutboundHandler.FML_MESSAGETARGET).set(FMLOutboundHandler.OutboundTarget.TOSERVER);
        channels.get(Side.CLIENT).writeOutbound(new PacketURLWriter(x, y, z, name, url, localName, size, color, author));
    }

    public static void sendURLWriteFromClient(int x, int y, int z, String name, String url, String localName, int size){
        channels.get(Side.CLIENT).attr(FMLOutboundHandler.FML_MESSAGETARGET).set(FMLOutboundHandler.OutboundTarget.TOSERVER);
        channels.get(Side.CLIENT).writeOutbound(new PacketURLWriter(x, y, z, name, url, localName, size));
    }

}

File: src/main/kotlin/com/codingforcookies/betterrecords/network/PacketHandler.kt
package com.codingforcookies.betterrecords.network

import com.codingforcookies.betterrecords.NETWORK_CHANNEL
import net.minecraftforge.fml.common.network.simpleimpl.IMessage
import net.minecraftforge.fml.common.network.simpleimpl.SimpleNetworkWrapper
import net.minecraftforge.fml.relauncher.Side.CLIENT
import net.minecraftforge.fml.relauncher.Side.SERVER

object PacketHandler {

    private val HANDLER = SimpleNetworkWrapper(NETWORK_CHANNEL)

    fun init() {
        HANDLER.registerMessage(PacketRecordPlay.Handler::class.java, PacketRecordPlay::class.java, 0, CLIENT)
        HANDLER.registerMessage(PacketRadioPlay.Handler::class.java, PacketRadioPlay::class.java, 1, CLIENT)
        HANDLER.registerMessage(PacketSoundStop.Handler::class.java, PacketSoundStop::class.java, 2, CLIENT)
        HANDLER.registerMessage(PacketWireConnection.Handler::class.java, PacketWireConnection::class.java, 3, SERVER)
    }

    fun sendToAll(msg: IMessage) {
        HANDLER.sendToAll(msg)
    }

    fun sendToServer(msg: IMessage) {
        HANDLER.sendToServer(msg)
    }
}